Kernel/Storage/SD/SDHostController.cpp

```diff
@@ -560,49 +595,312 @@ ErrorOr<void> SDHostController::transaction_control_with_data_transfer_using_the
+        //     Enable. Multi / Single Block Select and Block Count Enable are determined according to
+        //     Table 2-9.
+        //     If response check is enabled (Response Error Check Enable =1), set Response Interrupt
+        //     Disable to 1 and select Response Type R1 / R5
+        SD::Command command = {
+            .dma_enable = 1,
+            .block_counter = 0,
+            .auto_command = blocks_transferred > 1 ? SD::SendAutoCommand::Command12 : SD::SendAutoCommand::Disabled,
+            .direction = direction,
+            .multiblock = blocks_transferred > 1,
+            .response_type_r1r5 = 0,
+            .response_error_check = 0,
+            .response_interrupt_disable = 0,
+            .reserved1 = 0,
+            .response_type = SD::ResponseType::ResponseOf48Bits,
+            .sub_command_flag = 0,
+            .crc_enable = 1,
+            .idx_enable = 0,
+            .is_data = 1,
+            .type = SD::CommandType::Normal,
+            .index = direction == SD::DataTransferDirection::HostToCard ? (blocks_transferred > 1 ? SD::CommandIndex::WriteMultipleBlock : SD::CommandIndex::WriteSingleBlock)
+                                                                        : (blocks_transferred > 1 ? SD::CommandIndex::ReadMultipleBlock : SD::CommandIndex::ReadSingleBlock),
+            .reserved3 = 0
+        };
+
+        // (7) Set the value to the Command register.
+        //     Note: When writing to the upper byte [3] of the Command register, the SD command is issued
+        //     and DMA is started.
+        m_registers->transfer_mode_and_command = command.raw;
+
+        // FIXME: We dont actually enable response checking, and the specs flow chart
+        //        would have us enter block 11 after block 10
+        //        If we try to do steps 8-10 we hang up
+        //        So figure out if theres a spec bug here or response checking
+        //        refers to another flag than the commands response_error_check
+        //        Note: We use steps 11f, due to them being more openly error agnostic
+        //        (8) If response check is enabled, go to stop (11) else wait for the Command Complete Interrupt.
+        //        (9) Write 1 to the Command Complete in the Normal Interrupt Status register to clear this bit.
+        //        (10) Read Response register and get necessary information of the issued command
+
+        // (11) Wait for the Transfer Complete Interrupt and ADMA Error Interrupt.
+        // FIXME: Especially with big transfers this might timeout before the transfer is finished, although
+        //        No error has has happened
+        //        We should set this up so that it actually waits for the interrupts via a designated handler
+        if (!retry_with_timeout([this]() { return m_registers->interrupt_status.command_complete || m_registers->interrupt_status.adma_error; })) {
```

Comment:
You should be checking for `transfer_complete`, not `command_complete`

Reply:
Well that makes sense
also helped me find the bug that caused the above fixme